python/ray/tune/logger/aim.py

```diff
@@ -0,0 +1,192 @@
+import logging
+
+import numpy as np
+from typing import TYPE_CHECKING, Dict, Optional, List, Union
+
+from ray.tune.logger.logger import LoggerCallback
+from ray.tune.result import (
+    TRAINING_ITERATION,
+    TIME_TOTAL_S,
+    TIMESTEPS_TOTAL,
+)
+from ray.tune.utils import flatten_dict
+from ray.util.annotations import PublicAPI
+
+if TYPE_CHECKING:
+    from ray.tune.experiment.trial import Trial
+
+try:
+    from aim.sdk import Repo, Run
+except ImportError:
+    Repo, Run = None, None
+
+logger = logging.getLogger(__name__)
+
+VALID_SUMMARY_TYPES = [int, float, np.float32, np.float64, np.int32, np.int64]
+
+
+@PublicAPI
+class AimLoggerCallback(LoggerCallback):
+    """Aim Logger: logs metrics in Aim format.
+
+    Aim is an open-source, self-hosted ML experiment tracking tool.
+    It's good at tracking lots (thousands) of training runs, and it allows you to
+    compare them with a performant and well-designed UI.
+
+    Source: https://github.com/aimhubio/aim
+
+    Args:
+        repo: Aim repository directory or a `Repo` object that the Run object will
+            log results to. If not provided, a default repo will be set up in the
+            experiment directory (one level above trial directories).
+        experiment: Sets the `experiment` property of each Run object, which is the
+            experiment name associated with it. Can be used later to query
+            runs/sequences.
+            If not provided, the default will be the Tune experiment name set
+            by `RunConfig(name=...)`.
+        metrics: List of metric names (out of the metrics reported by Tune) to
+            track in Aim. If no metric are specified, log everything that
+            is reported.
+        **aim_run_kwargs: Additional arguments that will be passed when creating the
```

Comment:
I think we need to remove the asterisks for the rendering to work well

Reply:
Updated.